diff -p -up ./src/inp.h.tv ./src/inp.h
--- ./src/inp.h.tv	2009-12-03 14:39:10.000000000 +0100
+++ ./src/inp.h	2009-12-03 14:39:38.000000000 +0100
@@ -24,5 +24,5 @@ XTERN LINENUM input_lines;		/* how long 
 char const *ifetch (LINENUM, bool, size_t *);
 void get_input_file (char const *, char const *);
 void re_input (void);
-void reset_scan_input_vars PARAMS (());
+void reset_scan_input_vars ();
 void scan_input (char *);
