diff -p -up ./src/patch.c.tv ./src/patch.c
--- ./src/patch.c.tv	2009-11-24 18:43:32.000000000 +0100
+++ ./src/patch.c	2009-11-24 18:44:25.000000000 +0100
@@ -1208,7 +1208,7 @@ apply_hunk (struct outstate *outstate, L
 		return false;
 	    if (R_do_defines) {
 		if (def_state == OUTSIDE) {
-		    fprintf (fp, outstate->after_newline + not_defined,
+		    fprintf (fp, "%s", outstate->after_newline + not_defined,
 			     R_do_defines);
 		    def_state = IN_IFNDEF;
 		}
@@ -1236,7 +1236,7 @@ apply_hunk (struct outstate *outstate, L
 		    def_state = IN_ELSE;
 		}
 		else if (def_state == OUTSIDE) {
-		    fprintf (fp, outstate->after_newline + if_defined,
+		    fprintf (fp, "%s", outstate->after_newline + if_defined,
 			     R_do_defines);
 		    def_state = IN_IFDEF;
 		}
@@ -1304,7 +1304,7 @@ apply_hunk (struct outstate *outstate, L
 	    return false;
 	if (R_do_defines) {
 	    if (def_state == OUTSIDE) {
-		fprintf (fp, outstate->after_newline + if_defined,
+		fprintf (fp, "%s", outstate->after_newline + if_defined,
 			 R_do_defines);
 		def_state = IN_IFDEF;
 	    }
